File: autoit-mediapipe-com/src/binding/tasks/components/containers/category.cc
#include "binding/tasks/components/containers/category.h"

namespace mediapipe::tasks::components::containers {
	Classification ConvertCategoryToProto(Category* category) {
		Classification classification_proto;
		classification_proto.set_index(category->index);
		classification_proto.set_score(category->score);
		if (category->display_name) {
			classification_proto.set_display_name(*category->display_name);
		}
		if (category->category_name) {
			classification_proto.set_label(*category->category_name);
		}
		return classification_proto;
	}
}

File: src/declarations/tasks/components/containers/category.js
const ns = "mediapipe::tasks::components::containers";
const nsProgId = ns.replaceAll("::", ".");

module.exports = ({language}) => {
    const progid = `mediapipe.tasks.${ language }.components.containers.category`;

    return [
        // expose category properties like in mediapipe python
        [`${ progid }.`, "", ["/Properties"], [
            [`${ ns }::Category`, "Category", "", ["/R", "=this", "/S"]],
        ], "", ""],

        [`struct ${ nsProgId }.Category`, "", [`/progid=${ progid }.Category`, "/Simple", "/DC"], [
            ["int", "index", "-1", ["/RW"]],
            ["float", "score", "", ["/RW"]],
            ["std::optional<std::string>", "category_name", "std::nullopt", ["/RW"]],
            ["std::optional<std::string>", "display_name", "std::nullopt", ["/RW"]],
        ], "", ""],
    ];
};
